### ISSUE TITLE
Rewrite localization / strings generation script/parser

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#419]
### Problem

Our current script doesn't really work with `L` and `<i18n>`.

### Solution

- Rewrite what we have
- Make sure it gives errors whenever strings are fail to follow conventions (for example, if strings are concatenated using `+` or span multiple lines)
- Make sure it has a linter-mode where it just verifies the string conventions so it can be run as part of `grunt test` (i.e. without generating `.strings` files)
- Make sure it works with #417 
- Close #164 
- Add a way to ignore files using pattern matching (a kind of `.eslintignore` or `.gitignore` etc thing), so that files like the [design page](https://github.com/okTurtles/group-income-simple/pull/679/files/fe4e5cf13c179e2fd620ffb2560e769e246d98a3#r330817199) won't be parsed for strings

Useful tools:

- https://github.com/Cykelero/tasklemon/
- https://github.com/shelljs/shelljs
- https://rosie-lang.org/about/
- https://github.com/returntocorp/semgrep
- https://sap.github.io/chevrotain/docs/
- https://comby.dev/
- http://gema.sourceforge.net/new/index.shtml

EDIT: make sure it works with [Weblate](https://github.com/liberapay/liberapay.com#contributing-to-the-translations)

**Comment by taoeffect** [comment]
Closed by @SGrondin with his `strings` utility.
